B/Botan/build_tarballs.jl

```diff
@@ -0,0 +1,120 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "Botan"
+version = v"3.4.0"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource(
+        "https://botan.randombit.net/releases/Botan-$(version).tar.xz",
+        "71843afcc0a2c585f8f33fa304f0b58ae4b9c5d8306f894667b3746044277557",
+    ),
+    FileSource(
+        "https://github.com/joseluisq/macosx-sdks/releases/download/13.3/MacOSX13.3.sdk.tar.xz",
+        "518e35eae6039b3f64e8025f4525c1c43786cc5cf39459d609852faf091e34be",
+    ),
```

Comment:
Compilation with SDK 12 failed for me. If I remember correctly, the C++20 ranges library wasn't fully supported. But I can check again and try to fix this, if you think SDK 13 is too high.

Reply:
I still didn't get it to work with MacOSX SDK 12.3.

Alternatively, one could use the GCC of the build environments. The compilation works, only the linker warns about `libgcc.a` and `libemutls_w.a` not being built for macOS but for iOS. On Github's CI at least the resulting library for `x86 macOS` runs fine under Julia `1.6.3`, `1.10` and `nightly`. I can't check other Apple systems right now. Maybe `aarch64-apple` could make problems given that the GCC in the corresponding build environments was version 12 instead of 11 (in spite of `preferred_gcc_version = v"11"`).

Reply:
I keep it as is then, sticking with MacOSX SDK.